Dockerfile

```diff
@@ -10,6 +10,7 @@ COPY go.sum go.sum
 # Copy this, which should not change often; and, needs to be in place
 # before `go mod download`.
 COPY api/ api/
+COPY internal/ internal/
```

Comment:
This can go later, alongside `COPY controllers`; `api/` is copied early just because it has a go.mod and the next stage downloads modules (in the hope that layer can be cached if the go.mods haven't changed).